Sources/WebSocketKit/WebSocket.swift

```diff
@@ -334,7 +447,20 @@ public final class WebSocket: Sendable {
             }
         }
     }
-
+    
+    func unmasked(frame maskedFrame:WebSocketFrame) -> WebSocketFrame {
+        var unmaskedData = maskedFrame.data
+        unmaskedData.webSocketUnmask(maskedFrame.maskKey!)
+        return WebSocketFrame(fin: maskedFrame.fin,
+                              rsv1: maskedFrame.rsv1,
+                              rsv2: maskedFrame.rsv2,
+                              rsv3: maskedFrame.rsv3,
+                              opcode: maskedFrame.opcode,
+                              maskKey: maskedFrame.maskKey,//should this be nil
+                              data: unmaskedData,
+                              extensionData: maskedFrame.extensionData)
+    }
```

Comment:
How many copies of this logic do we need?

Reply:
Not sure.

Reply:
I dont see what functionality is copied. that unmask is on ByteBuffer and a new frame still needs to be made as far as I can tell. if not careful and explicit unmask will just mask your data again so creating a new frame helps make it obvious that the data was already unmasked I suppose.